### PR TITLE
feat: monitor work in urgency

### PR DESCRIPTION
**Pull request by MartinBB360**
### Context
https://trello.com/c/LGUF7nFF/384-%F0%9F%9A%91-monitor-the-occurences-of-working-out-of-process

### Specs

Anyone can access : https://360learning.github.io/trelloEngWorkInUrgencyDashboard.html
Here is the pipeline:
- fill Trello API Key + token
- provide a quarter
- provide a squad name (or all squads)
- get a report

_Notes_: since the repo is public, I cannot put sensitive information in the code, such as squad names.
That's why I retrieve the squad names from the list's name.
And I only store the list ids, that cannot be exploited without proper access to our Trello env.

### CODE REVIEW

**Comment by MartinBB360** [comment]
@colingrodecoeur360 Hi !
I finished the code for the work in urgency monitoring.
I did not split it into several commits but would you validate the pipeline detailed in the PR desc ?
The challenge was not to store any sensitive info in the code.
The only drawback in the current version is that when we will want to add a new squad to the monitoring, it will require a PR to update the list of ids. It's ok imo but we could also store this list in a unique Trello card, and access it via the api.

Wdyt ?

If needed, tell me and we can plan a 5min sync for me to demo it 😉

**Comment by colingrodecoeur360** [comment]
@MartinBB360 It looks good to me, thanks for this work 👍 

> It's ok imo but we could also store this list in a unique Trello card, and access it via the api.

I think it's fine like this; if the squad list changes at some point, updating the list here won't take much more time than updating a Trello card, and it won't happen often.

**Comment by MartinBB360** [comment]
@LucVidal360 @arslane-boukrif 
Hi !
Here is a PR I merged to set up a new process to help squads monitor their tasks that are done in urgency / out of the process.
This is not a usual dev repo, so I went fast and only asked for a review from Colin that had already implemented things here.
The communication will soon be done, but you can find a draft [here](https://trello.com/c/LGUF7nFF/384-%F0%9F%9A%91-monitor-the-occurrences-of-working-out-of-process#comment-65dda8b613d404de5443a6f5).
This is a task I do as part of my TL role, but I thought you might be interested 😉

File: trelloEngWorkInUrgencyDashboard.html [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+                        <template v-else>Access dashboard</template>
+                        </button>
+                        <br>
+
+                        <div v-if="squadNames">
+
+                            <h2>Settings</h2>
+
+                            <div class="form-field">
+                                <label for="quarter" class="field-name">Quarter</label>
```
Comment (LucVidal360) [inline trelloEngWorkInUrgencyDashboard.html:62]: @MartinBB360 The `for` attribute here is useless, has it doesn't correspond to the `id` attribute of a corresponding form input.

Likewise for the second `select`.

File: trelloEngWorkInUrgencyDashboard.html [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+                                <select name="quarters" v-model="options.quarter">
+                                    <option value="Q4 2023">Q4 2023</option>
+                                    <option value="Q1 2024">Q1 2024</option>
+                                    <option value="Q2 2024">Q2 2024</option>
+                                    <option value="Q3 2024">Q3 2024</option>
+                                    <option value="Q4 2024">Q4 2024</option>
+                                    <option value="Q1 2025">Q1 2025</option>
+                                    <option value="Q1 2025">Q2 2025</option>
+                                    <option value="Q1 2025">Q3 2025</option>
+                                    <option value="Q1 2025">Q4 2025</option>
```
Comment (LucVidal360) [inline trelloEngWorkInUrgencyDashboard.html:72]: @MartinBB360 This is kind of hard-coded :sweat_smile:

**Review by LucVidal360 — COMMENTED** [review]
@MartinBB360 Interesting :pray:

File: scripts/trelloEngWorkInUrgencyDashboard.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+            trelloOAuth1: ""
+        },
+        urgencyCardsBySquadName: {},
+        squadNames: null,
+        listIdBySquadName: {},
+        reportBuilt: false,
+        reportBuildForAllSquads: false
+    },
+    computed: {
+        areCredentialsValid() {
```
Comment (arslane-boukrif) [inline scripts/trelloEngWorkInUrgencyDashboard.js:29]: [Nit] Would rename to `areCredentialsFilled` as you don't check if they are valid here.

File: trelloEngWorkInUrgencyDashboard.html
Comment (arslane-boukrif) [inline trelloEngWorkInUrgencyDashboard.html:1]: [Question] Wouldn't it be simpler to fetch all the lists/cards and then filter based on the results?

**Comment by MartinBB360** [comment]
@LucVidal360 @arslane-boukrif 
Thanks for your feedbacks: I treated them here : https://github.com/360Learning/360learning.github.io/pull/25

File: trelloEngWorkInUrgencyDashboard.html (hunk 1 above)
Comment (MartinBB360) [reply trelloEngWorkInUrgencyDashboard.html:62]: @LucVidal360 Fixed here https://github.com/360Learning/360learning.github.io/pull/25/commits/0412146ed327c0ac042c6a140b7f7aa65c7826e1

File: trelloEngWorkInUrgencyDashboard.html (hunk 2 above)
Comment (MartinBB360) [reply trelloEngWorkInUrgencyDashboard.html:72]: @LucVidal360 Indeed 😅 Fixed here https://github.com/360Learning/360learning.github.io/pull/25/commits/b33c3586a1853578776fd1ec19f66ffb5508777b

File: scripts/trelloEngWorkInUrgencyDashboard.js (hunk 3 above)
Comment (MartinBB360) [reply scripts/trelloEngWorkInUrgencyDashboard.js:29]: @arslane-boukrif Fixed here https://github.com/360Learning/360learning.github.io/pull/25/commits/a8b66723cb07a2b7b1bbe13e23882f2582bc1c49

File: trelloEngWorkInUrgencyDashboard.html
Comment (MartinBB360) [reply trelloEngWorkInUrgencyDashboard.html:1]: @arslane-boukrif I am not sure to understand the suggestion. Could you detail it ?

Comment (arslane-boukrif) [reply trelloEngWorkInUrgencyDashboard.html:1]: @MartinBB360 From my understanding here we do the following steps:
- Ask for trello api key and token
- Fetch squad names
- Ask for squad/quarter selection
- Fetch cards in selected squad lists
- Filter by quarter

And I'm wondering if it wouldn't be simpler to:
- Ask for trello api key and token
- Fetch cards in all squad lists
- Display all squad lists by default
- Add filters based on fetched data (eg: compute the quarters dropdown filter based on all existing quarters in cards)

Hope it is clearer 😅

Comment (MartinBB360) [reply trelloEngWorkInUrgencyDashboard.html:1]: @arslane-boukrif Understood, thanks !
This is a nice idea, similar to what Colin suggested [here](https://trello.com/c/LGUF7nFF/384-%F0%9F%9A%91-monitor-the-occurrences-of-working-out-of-process#comment-65e06f59c755e36be2d23669).
I answered [here](https://trello.com/c/LGUF7nFF/384-%F0%9F%9A%91-monitor-the-occurrences-of-working-out-of-process#comment-65e088f45c6f03b1bbfdaf1c):
> Let’s consider the current version a V0, that is working, and we can now wait for more feedbacks. For example, we’ll know if squads prefer monitoring things directly in their board (in that case the dashboard would only be used by Fabrice).
Then, if a V1 is needed, I think I’ll call for a frontend volunteer that will do something way better than me, and way faster 😅

I logged these ideas in the "improvement ideas" of the card.
Thanks again for the feedback 🙏

Comment (arslane-boukrif) [reply trelloEngWorkInUrgencyDashboard.html:1]: @MartinBB360 Great! Thanks for the taking the time 🙏

**Pull request merged**